src/Statistics.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Statistics
```

Comment:
This is the file where all the functionality should go in. I understand you might have done this by example, but the whole point of creating this file is to have all related functionality in it. Move all the statistics code here, otherwise, we will have a ton of code conflicts.